lib/internal/modules/esm/hooks.js

```diff
@@ -528,7 +535,10 @@ class HooksProxy {
         debug('wait for signal from worker');
         AtomicsWait(this.#lock, WORKER_TO_MAIN_THREAD_NOTIFICATION, 0);
         const response = this.#worker.receiveMessageSync();
-        if (response.message.status === 'exit') { return; }
+        if (response.message.status === 'exit') {
+          process.exit(response.message.body);
+          return;
+        }
```

Comment:
Let's not change that, otherwise it's going to conflict with the `globalPreload` fix.
```suggestion
        if (response.message.status === 'exit') { return; }
```

Reply:
I will test this locally but I have a hunch this will cause some tests not to pass.

Reply:
Apparently it's fine 🤷 I guess another fixup or refactor has ameliorated this problem for now. Pushing proposed change.